lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -178,6 +189,9 @@ def compile_module(tree, name, constant)
               RBI::Module.new(name)
             end
 
+          comments = documentation_comments(name)
+          scope.comments.concat(comments) unless comments.empty?
+
```

Comment:
Same here, you can move those lines before the initializations of the `Class` and `Module` and pass the comments to the `new` call.